compiler/src/scala/quoted/runtime/impl/QuotesImpl.scala

```diff
@@ -253,6 +253,12 @@ class QuotesImpl private (using val ctx: Context) extends Quotes, QuoteUnpickler
         (cdef.name.toString, cdef.constructor, cdef.parents, cdef.self, rhs.body)
 
       def module(module: Symbol, parents: List[Tree /* Term | TypeTree */], body: List[Statement]): (ValDef, ClassDef) = {
+        xCheckMacroAssert(parents.nonEmpty && !parents.head.tpe.dealias.classSymbol.is(dotc.core.Flags.Trait), "First parent must be a class")
+        xCheckMacroAssert(
```

Comment:
We should factor out the common logic with https://github.com/scala/scala3/pull/18935/files#diff-0819ff2757ddb9b8950698b5e502c765cc74ff4ddbd0a5a729b71d9e3eb8b217R554-R563